crates/router/src/core/webhooks.rs

```diff
@@ -385,24 +383,27 @@ pub async fn disputes_incoming_webhook_flow<W: types::OutgoingWebhookType>(
         )
         .await?;
         let disputes_response = Box::new(dispute_object.clone().foreign_into());
-        let event_type: enums::EventType = dispute_object
-            .dispute_status
-            .foreign_try_into()
-            .into_report()
-            .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
-            .attach_printable("failed to map dispute status to event type")?;
-        create_event_and_trigger_outgoing_webhook::<W>(
-            state,
-            merchant_account,
-            event_type,
-            enums::EventClass::Disputes,
-            None,
-            dispute_object.dispute_id,
-            enums::EventObjectType::DisputeDetails,
-            api::OutgoingWebhookContent::DisputeDetails(disputes_response),
-        )
-        .await?;
-        metrics::INCOMING_DISPUTE_WEBHOOK_MERCHANT_NOTIFIED_METRIC.add(&metrics::CONTEXT, 1, &[]);
+        let event_type: enums::EventType = dispute_object.dispute_status.foreign_into();
+
+        // If event is NOT an UnsupportedEvent, trigger Outgoing Webhook
+        if !matches!(event_type, enums::EventType::UnsupportedEvent) {
+            create_event_and_trigger_outgoing_webhook::<W>(
+                state,
+                merchant_account,
+                event_type,
+                enums::EventClass::Disputes,
+                None,
+                dispute_object.dispute_id,
+                enums::EventObjectType::DisputeDetails,
+                api::OutgoingWebhookContent::DisputeDetails(disputes_response),
+            )
+            .await?;
+            metrics::INCOMING_DISPUTE_WEBHOOK_MERCHANT_NOTIFIED_METRIC.add(
```

Comment:
Add metrics to also catch hyperswitch-unsupported events. We can take this up later. For now, please create an issue.